.circleci/config.yml

```diff
@@ -6,8 +6,63 @@ jobs:
     steps:
       - checkout
       - run:
-          name: Build and Test
+          name: Prepare
+          command: npm install
+      - run:
+          name: Build
+          command: npm run build
+      - run:
+          name: Unit Test (with coverage)
+          command: npm run coverage
+      - run:
+          name: Run Chrome tests
+          command: npm run dev & npm run selenium-chrome
+      - run:
+          name: Run Firefox tests
+          command: npm run dev & npm run selenium-firefox
+      - run:
+          name: Run IE tests
+          command: npm run dev & npm run selenium-ie
+      - run:
+          name: Run Safari tests
+          command: npm run dev & npm run selenium-safari
+      - run:
+          name: Run Edge tests
+          command: npm run dev & npm run selenium-edge
+      - run:
+          name: Prepare demo
+          command: ./scripts/demo.sh
+      - persist_to_workspace:
+          root: .
+          paths: ./demo
+  update-demo:
+    docker:
+      - image: node:8.10.0
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - add_ssh_keys:
+          fingerprints:
+              - "c3:03:93:79:5d:c7:23:b8:4a:4d:5d:41:e3:38:3f:84"
+      - run:
+          name: Prepare
           command: |
-            npm install
-            npm run build
-            npm run coverage
+            npm install -g --silent gh-pages
+            git config user.email "ci-build@wikimedia.org"
+            git config user.name "ci-build"
+      - run:
+          name: Deploy demo
+          command: |
+            gh-pages --dist demo
+workflows:
+  version: 2
+  build:
+    jobs:
+      - build
+      - update-demo:
+          requires:
+            - build
+          filters:
+            branches:
+              only: master
```

Comment:
if this deploys only on master maybe we can create a new demo page just for tests that deploys on every build?

Reply:
Ideally the tests should run against the local version of the code

Reply:
The local connection is not reliable enough for us to trust it IMO.
Deploying the code on a test page should be fairly simple, no?

Reply:
can we have a test environment to run the tests against?

Reply:
we use github gh-pages serve as the master environment for the demo purpose, unfortunately it doesn't support extra environment. We would need to use local version for the CI test.

Reply:
😢

Reply:
All PRs need to be tested independently so we would need a distinct environment for each. And we need to be able to clean them up at some point. It's possible but I'm not sure what a simple solution could be.

Testing against local is so much better and simpler. CircleCI has a large collection of images with different browsers. Maybe worth looking if we can find some IEs in there.

Also, when you say the local connection is not reliable, what do you mean, what do you see? Maybe there's something we can do to help it.

Reply:
I've opened a support ticket with cbt, it's a problem on their side (they open a tunnel to where the code is running and it doesn't seem to work very well/slow).

If we can run the tests inside circleci would be perfect, I just need to know the exact version of the browser that is running in the image.